lib/global/index.js

```diff
@@ -6,11 +6,36 @@
 'use strict'; /* eslint strict:0 */
 
 var getGlobal = function () {
-  if (typeof globalThis !== 'undefined') { return globalThis; }
-  if (typeof window !== 'undefined') { return window; }
-  if (typeof self !== 'undefined') { return self; }
-  if (typeof global !== 'undefined') { return global; }
-  throw new Error('unable to locate global object');
+  var globalObj = (function () {
+    if (typeof globalThis !== 'undefined') { return globalThis; }
+    if (typeof window !== 'undefined') { return window; }
+    if (typeof self !== 'undefined') { return self; }
+    if (typeof global !== 'undefined') { return global; }
+    throw new Error('unable to locate global object');
+  })()
+  /*
+    The below code was added in case there is pollution in global namespace
+    windows object during transpilation with __esModule being set. Below code would support
+    global import in all bundle use cases
+  */
+  if (globalObj && globalObj.__esModule) {
+    const override = {
+      get: function (target, prop) {
+        if (prop === 'default') {
+          return target
+        }
+
+        return target[prop]
+      },
+
+      set: function (target, prop, value) {
+        target[prop] = value
+      },
+    };
```

Comment:
Return a `Reflect` function from the get trap
And no need to trap the `set` method, let it be transparent

```javascript
get: function (target, prop, receiver) {
  if (prop === 'default') {
    return target
  }
  
  return Reflect.get(target, prop, receiver);
},
```

Reply:
> Return a `Reflect` function from the get trap to preserve the correct scope. More details in https://javascript.info/proxy#reflect And no need to trap the `set` method, let it be transparent
> 
> ```js
> get: function (target, prop, receiver) {
>   if (prop === 'default') {
>     return target
>   }
>   
>   return Reflect.get(target, prop, receiver);
> },
> ```

This is done, 
Reflect.get(target, prop, receiver) - Throws a "illegal invocation javascript". This is due to reference change between proxy and actual windows when we try to invoce it. Replaced with Reflect.get(receiver, prop, target)

setter also same, used reflect here as well. Got  ""illegal invocation javascript"" error here as well without adding setter